packages/mui-material/src/Select/SelectInput.js

```diff
@@ -418,10 +418,10 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
 
   if (computeDisplay) {
     if (multiple) {
-      if (value.length === 0) {
-        display = '';
+      if (displayMultiple.length === 0) {
+        display = null;
       } else {
-        display = displayMultiple.reduce((prev, curr) => [prev, ', ', curr]);
+        display = displayMultiple.reduce((prev, curr) => [...prev, ', ', curr]);
```

Comment:
to keep the React tree simple (flat)

Reply:
This leads to two bugs: (1) `TypeError: prev is not iterable` and (2) value gets split (see screenshot below)

![Screenshot 2021-12-10 at 09 10 04](https://user-images.githubusercontent.com/32841130/145548121-ee0ca99a-0128-47c5-b63d-61eb22493a6e.png)

Reply:
Arf, maybe we could reuse https://github.com/mui-org/material-ui/blob/14dd1128f3fc8355a87191fea5fcd48e55182161/packages/mui-material/src/Stack/Stack.js#L21 in this case.

Reply:
Yes, I agree. We shouldn't have a nested array especially when we don't need to.

I made a [commit](https://github.com/mui-org/material-ui/pull/30135/commits/05c9338247f45788798122a8c154a070178d2c3d) here.